activerecord/test/cases/adapters/postgresql/postgresql_adapter_test.rb

```diff
@@ -589,6 +589,15 @@ def test_allowlist_of_warnings_to_ignore
         end
       end
 
+      def test_allowlist_of_warning_codes_to_ignore
+        with_example_table do
+          with_db_warnings_action(:raise, ["01000"]) do
+            result = @connection.execute("do $$ BEGIN RAISE WARNING 'PostgreSQL SQL warning'; END; $$")
```

Comment:
> The test in its current form doesn't really test the allowlist because the warning won't match the provided code anyways. 

how so? I added a debug statement to check the warning object returned and this is what I got

```
  1199|         def warning_ignored?(warning)
=>1200|           require "debug"; debugger
  1201|           ActiveRecord.db_warnings_ignore.any? do |warning_matcher|
  1202|             warning.message.match?(warning_matcher) || warning.code.to_s.match?(warning_matcher)
  1203|           end
  1204|         end
=>#0    ActiveRecord::ConnectionAdapters::AbstractAdapter#warning_ignored?(warning=#<ActiveRecord::SQLWarning: PostgreSQL SQ...) at ~/nickborromeo/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:1200
  #1    ActiveRecord::ConnectionAdapters::PostgreSQL::DatabaseStatements#warning_ignored?(warning=#<ActiveRecord::SQLWarning: PostgreSQL SQ...) at ~/nickborromeo/rails/activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb:201
  # and 39 frames (use `bt' command for all frames)
(rdbg) warning
"#<ActiveRecord::SQLWarning: PostgreSQL SQL warning>"
(ruby) warning.message
"PostgreSQL SQL warning"
(ruby) warning.code
"01000"
```

Happy to add a more real world example though if you have something in mind.

Reply:
Oh that's my bad! I didn't realize that if you manually raised a PG warning it would assign it a code. (The [PG errcode](https://www.postgresql.org/docs/current/errcodes-appendix.html) docs clearly outline 01000 as the base warning code 🤦‍♀️). Ignore me! 😄